lib/smol_str/src/lib.rs

```diff
@@ -875,17 +875,17 @@ impl SmolStrBuilder {
 
     /// Builds a [`SmolStr`] from `self`.
     #[must_use]
-    pub fn finish(&self) -> SmolStr {
-        SmolStr(match &self.0 {
-            &SmolStrBuilderRepr::Inline { len, buf } => {
+    pub fn finish(self) -> SmolStr {
+        SmolStr(match self.0 {
+            SmolStrBuilderRepr::Inline { len, buf } => {
                 debug_assert!(len <= INLINE_CAP);
                 Repr::Inline {
                     // SAFETY: We know that `value.len` is less than or equal to the maximum value of `InlineSize`
                     len: unsafe { InlineSize::transmute_from_u8(len as u8) },
                     buf,
                 }
             }
-            SmolStrBuilderRepr::Heap(heap) => Repr::new(heap),
+            SmolStrBuilderRepr::Heap(heap) => Repr::Heap(Arc::from(heap.into_boxed_str())),
```

Comment:
This will allocate more, not less. `Arc::from` will allocate anyway, so this just has the extra allocation for `into_boxed_str()`.